src/main/java/by/custom/utilcalculator/service/BotFieldsManager.java

```diff
@@ -0,0 +1,100 @@
+package by.custom.utilcalculator.service;
+
+import by.custom.utilcalculator.directory.BotEntity;
+import by.custom.utilcalculator.directory.resources.Commands;
+import by.custom.utilcalculator.directory.steps.*;
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
+import org.telegram.telegrambots.meta.api.objects.Message;
+
+public class BotFieldsManager {
+    private final BotEntity botEntity;
+    private final MessagesManager messagesManager;
+
+    public BotFieldsManager() {
+        botEntity = BotEntity.getInstance();
+        messagesManager = MessagesManager.getInstance();
+    }
+
+    public static BotFieldsManager getInstance() {
+        return BotFieldsManagerHolder.BOT_FIELDS_MANAGER;
+    }
+
+    public SendMessage setCarOriginFieldAndGetNextMessageForUser(Message inputMessage) {
+        cleanStepsAfterCurrent(1);
+        switch (inputMessage.getText()) {
+            case Commands.EAES -> botEntity.setCountryOrigin(CountryOrigin.EAES);
+            case Commands.OTHER_COUNTRIES -> botEntity.setCountryOrigin(CountryOrigin.OTHER);
+        }
+        return messagesManager.createSendMessage(inputMessage.getChatId().toString(),
+                botEntity.getCountryOrigin(), botEntity.getOwnersType(), botEntity.getTypeOfEngine(), botEntity.getVolumeOfEngine(), botEntity.getCarAge());
+    }
+
+    public SendMessage setTypeOfOwnerFieldAndGetNextMessageForUser(Message inputMessage) {
+        cleanStepsAfterCurrent(2);
+        switch (inputMessage.getText()) {
+            case Commands.JURIDICAL_PERSON -> botEntity.setOwnersType(OwnersType.JURIDICAL);
+            case Commands.PHYSICAL_PERSON -> botEntity.setOwnersType(OwnersType.PHYSICAL);
+        }
+        return messagesManager.createSendMessage(inputMessage.getChatId().toString(),
```

Comment:
Please, move `createSendMessage(` to MessageRouter

Reply:
I think, we don't need this method anymore. I have removed it at all. Now we can call MessagesManagers methods from BotFieldsManager class. We don't need this wrapper createSendMessage() anymore, I guess. All it's logic is in the MessageRouter router and MessageManager. So I removed it.